group-coordinator/src/main/java/org/apache/kafka/coordinator/group/GroupMetadataManager.java

```diff
@@ -3388,11 +3412,18 @@ public void replay(
     ) {
         String groupId = key.groupId();
         String memberId = key.memberId();
-        ModernGroup<?> group = getOrMaybeCreatePersistedConsumerGroup(groupId, false);
 
         if (value != null) {
+            ConsumerGroup group = getOrMaybeCreatePersistedConsumerGroup(groupId, true);
```

Comment:
To trigger issue#2, we would need a new ConsumerGroupMemberMetadata record for an existing member, is this correct?

Reply:
Right.